### PR TITLE
Flightphase otits

### PR DESCRIPTION
**Pull request by celery6**
ajsdlfakhsdlgkasdjf;lsdkjgsdf
tested on nucleo thing, seems to work.
also seems to fail correctly if i mess up the flightphase logic on purpose.

### CODE REVIEW

File: STM32Cube/Tasks/flight_phase.c [hunk 1]
```diff
@@ -8,53 +8,113 @@
...
 #define RECOVERY_DEPLOYMENT (xEventGroupGetBits(flightPhaseEventsHandle) & RECOVERY_DEPLOYMENT_BIT)
 
+static Otits_Result_t test_flightPhase() {
+    Otits_Result_t res;
+    float time = millis_();
 
-void flightPhaseInit()
-{
+    // check that millis matches tick count to a reasonable degree TODO: is this how timer works lmao
+    if (time - xTaskGetTickCount() > 2) {
```
Comment (celery6) [inline STM32Cube/Tasks/flight_phase.c:30]: is this interpreting the function of millis_() correctly

File: STM32Cube/Tasks/flight_phase.c [hunk 2]
```diff
@@ -8,53 +8,113 @@
...
+        return res;
+    }
+
+    // must be in prelaunch if inj hasn't opened yet
+    if (!INJ_OPEN) {
+        if (flight_state != PHASE_PRELAUNCH) {
+            res.info = "prelaunch not detected";
+            res.outcome = TEST_OUTCOME_FAILED;
+            return res;
+        } else if (time > 30000) { // check inj_open hasn't gotten stuck TODO: what should this number be ??
```
Comment (celery6) [inline STM32Cube/Tasks/flight_phase.c:42]: the thought with this is to check if CAN received an inj_open msg and set the bit correctly. is this worth doing? is there a max time thats actually reasonable to test with?

File: STM32Cube/Tasks/flight_phase.c [hunk 3]
```diff
@@ -8,53 +8,113 @@
...
+
+bool flightPhaseInit() {
 	flightPhaseEventsHandle = xEventGroupCreate();
+	if (flightPhaseEventsHandle == NULL) return false;
 	xEventGroupClearBits(flightPhaseEventsHandle, 0xFFFF); //clear out the enitire group since API doesn't specify if values are initialized to 0
+	if (!otitsRegister(test_flightPhase, TEST_SOURCE_FLIGHT_PHASE, "phaseCheck")) return false;
+	return true;
 }
 
-void flightPhaseTask(void * argument)
```
Comment (celery6) [inline STM32Cube/Tasks/flight_phase.c:30]: the changes here is just reformatting, changing osdelay to vtaskdelay, and adding log msgs.

File: STM32Cube/Tasks/flight_phase.c (hunk 1 above)
Comment (Joe-Joe-Joe-Joe) [reply STM32Cube/Tasks/flight_phase.c:30]: Yes. I mean technically if you change the timebase of freertos it isn't, but I think its fair to assume we use 1000 ticks/second

File: STM32Cube/Tasks/flight_phase.c (hunk 2 above)
Comment (Joe-Joe-Joe-Joe) [reply STM32Cube/Tasks/flight_phase.c:42]: I don't think there is a reasonable limit. The board will probably be on for an hour or two before the injector valve opens

**Review by Joe-Joe-Joe-Joe — APPROVED** [review]
lgtm

**Pull request merged**